extensions-core/mysql-metadata-storage/src/main/java/io/druid/metadata/storage/mysql/MySQLConnector.java

```diff
@@ -163,17 +162,19 @@ protected int getStreamingFetchSize()
   @Override
   public boolean tableExists(Handle handle, String tableName)
   {
-    // ensure database defaults to utf8, otherwise bail
-    boolean isUtf8 = handle
-        .createQuery("SELECT @@character_set_database = 'utf8'")
-        .map(BooleanMapper.FIRST)
+    String databaseCharset = handle
+        .createQuery("SELECT @@character_set_database")
+        .map(StringMapper.FIRST)
         .first();
 
-    if (!isUtf8) {
+    if (!databaseCharset.matches("utf.*")) {
```

Comment:
Is the result ever cased differently? Do we need a to lowercase?

Reply:
Don't this also match a bunch of other things like utf16?

Reply:
Don't you think lowercase is okay, if it has been until now? But yes, it should be `uft8.*`. I'll fix it.